src/src/Helpers/Networking.cpp

```diff
@@ -1682,6 +1682,115 @@ int http_authenticate(const String& logIdentifier,
       }
     }
     #endif
+
+    # if FEATURE_OPENMETEO_EVENT
+
+    // Generate an event with the response of an open-meteo request.
+    // Example command:
+    // sendtohttp,api.open-meteo.com,80,"/v1/forecast?latitude=52.52&longitude=13.41&current=temperature_2m,weather_code,wind_speed_10m&daily=temperature_2m_max,temperature_2m_min&forecast_days=1"
+    // No need for an api key and it is free (daily requests are limited to 10,000 in the free version)
+    // Visit the URL (https://open-meteo.com/en/docs) and build your personal URL by selecting the location and values you want to receive.
+    // Supported variable kinds are current, hourly, daily!
+    // In rules you can grep the reply by the kind of weather variables with "On Openmeteo#<type> Do ..."
+    // e.g. "On Openmeteo#current Do ..."
+    // Note: hourly and daily results are arrays which can become very long.
+    // Best to make seperate calls. Especially for hourly results.
+
+    if ((httpCode == 200) && equals(host, F("api.open-meteo.com")))
+    {
+      // Length of the response is limited to 5000 characters
+      // ToDo: What would be right size?
+      if (uri.length() > 5000) {
+        addLog(LOG_LEVEL_ERROR, F("Response exceeds 5000 characters"));
+      }
+      else {
+        const String str = http.getString();
+
+        auto processAndQueueParams = [](const String& url, const String& str, const String& eventName) {
+            // Extract the parameters from the URL
+            int start = url.indexOf(eventName + '=');
+            if (start == -1) {
+                return; // No parameters found for the given eventName
+            }
+            start += eventName.length() + 1;
+            const int end = url.indexOf('&', start);
+            const String params = (end == -1) ? url.substring(start) : url.substring(start, end);
+
+            if (!params.isEmpty()) {
+                String keys[20];

```

Comment:
Please try to avoid using so called magic numbers.
Better to use a `#define WEATHER_KEYS_MAX 20` or something similar, so it can be managed from 1 place in the source.
An alternative could be to use a `std::vector` instead of an array, so it will grow as needed 😉 

NB: Common exceptions for magic numbers are 0 and 1.

Reply:
what about: 
```
if (uri.length() > 5000) {
        addLog(LOG_LEVEL_ERROR, F("Response exceeds 5000 characters"));
      }
      else {
```
This is not really necessary right?

Reply:
Well, the max. size could very well be dependent on the MCU generation, for ESP8266 is might even fail on smaller requests...

Reply:
but as soon as i call `uri.length() ` it will be in ram and therefore i prevented nothing with the statement? At least that's what I imagine

Reply:
If it is too large to fit in memory, the ESP will crash before reaching your check 💥

Reply:
so i could leave it as a warning message process the rest anyway when esp is still running at this point :)